onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -389,15 +403,15 @@ Status CUDAExecutionProvider::Sync() const {
 Status CUDAExecutionProvider::OnRunStart(const onnxruntime::RunOptions& /*run_options*/) {
   // always set CUDA device when session::Run() in case it runs in a worker thread
   CUDA_RETURN_IF_ERROR(cudaSetDevice(GetDeviceId()));
-  if (IsGraphCaptureEnabled() && GetPerThreadContext().IsGraphCaptureAllowed() && !GetPerThreadContext().IsGraphCaptured()) {
+  if (IsGraphCaptureEnabled() && !GetPerThreadContext().IsGraphCaptureSkippedOnRun() && GetPerThreadContext().IsGraphCaptureAllowed() && !GetPerThreadContext().IsGraphCaptured()) {
```

Comment:
Is it possible to put the logic to determine if graph capture is to be skipped (or not) in this Run() to IsGraphCaptureAllowed() ? It would simplify understanding this complex if criteria.

Reply:
I think that IsGraphCaptureAllowed() has a different meaning to !IsGraphCaptureSkippedOnRun(). e.g in inference_session.cc we cannot use IsGraphCaptureAllowed() to control if executing another dry run (https://github.com/microsoft/onnxruntime/blob/a9568935a52b3d51ec802a4ab89ab3852129fc1e/onnxruntime/core/session/inference_session.cc#L2564)